source/common/stats/stats_impl.cc

```diff
@@ -103,13 +105,18 @@ TagExtractorPtr TagExtractorImpl::createTagExtractor(const std::string& name,
     throw EnvoyException(fmt::format(
         "No regex specified for tag specifier and no default regex for name: '{}'", name));
   }
-  return TagExtractorPtr{new TagExtractorImpl(name, regex)};
+  return TagExtractorPtr{new TagExtractorImpl(name, regex, substr)};
 }
 
 bool TagExtractorImpl::extractTag(const std::string& stat_name, std::vector<Tag>& tags,
                                   IntervalSet<size_t>& remove_characters) const {
-
   PERF_OPERATION(perf);
+
+  if (!substr_.empty() && stat_name.find(substr_) == std::string::npos) {
```

Comment:
Can you add a simple test case to isolate this new portion of the logic and ensure it works as expected.  The code is already covered by the regex test cases, but adding a separate test will isolate the functionality from the tests that ensure that the regexes work correctly.  See tests in https://github.com/envoyproxy/envoy/blob/master/test/integration/stats_integration_test.cc and https://github.com/envoyproxy/envoy/blob/master/test/common/stats/stats_impl_test.cc#L88 for other tests that test specific parts of the extraction process.

Reply:
done.  Broke out substrMismatch as a separate method to make it easier to test.